### PR TITLE
Microphone button status change on narrator

### LINKED ISSUE #1409: [MAS4.1.2] Does not provide the information about the 'Mic' button that Mic has been activated.
**Actual Result:**
Does not give the information about the Mic button that Mic has been activated.
Narrator is reading complementary landmark after enable the 'Mic' button.

**Expected Result:**
Should give the information about the Mic button that Mic has been activated.

Repro Steps:

1. Open URL https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmicrosoft.github.io%2FBotFramework-WebChat%2Fspeech-cognitive-services-bing-speech%2F%3Fs%3D7c7dacfe04044f4f99a1edcf9b0e2ab9&data=02%7C01%7Cv-saugup%40microsoft.com%7C0f6f5e63167449daa46d08d65570f676%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636790341026155420&sdata=gcFq8JsflCLfUKAyw8f2A19aJcEQEFxGzjASMK9S59A%3D&reserved=0 in Edge browser.
2. Notice 'Upload File' and 'Mic' button icons present in bottom of screen.
3. Open Narrator (Ctrl.+Win. Key+ Enter)
4. Navigate to mic button by using Tab key and press Enter to activate it.
5. Mic button will activate.
6. Observe that after activate the 'Mic' button, narrator is reading complementary landmark.

**User Impact:**
Screen reader users may get confuse if not give the complete information about the activated control.

**Test Environment:**
**OS:** Windows 10
**OS Build:** 17134.345
**OS Version:** 1803
**Browser:** Edge
**Tool:** Narrator

**Attachment:**
[MAS4.1.2_not give the information that mic button has been activated..zip](https://github.com/Microsoft/BotFramework-WebChat/files/2628446/MAS4.1.2_not.give.the.information.that.mic.button.has.been.activated.zip)

### PR DESCRIPTION
**Pull request by corinagum**
Fix #1409

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+- Fix [#1409](https://github.com/Microsoft/BotFramework-WebChat/issues/1409). Patched something, by [@corinagum](https://github.com/corinagum) in PR [#1490](https://github.com/Microsoft/BotFramework-WebChat/pull/1490)
```
Comment (compulim) [inline CHANGELOG.md]:
```suggestion
- Fix [#1409](https://github.com/Microsoft/BotFramework-WebChat/issues/1409). Added microphone status as screen reader only text, by [@corinagum](https://github.com/corinagum) in PR [#1490](https://github.com/Microsoft/BotFramework-WebChat/pull/1490)
```

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 2]
```diff
@@ -12,7 +12,17 @@ import MicrophoneIcon from './Assets/MicrophoneIcon';
 const { DictateState } = Constants;
 
 const ROOT_CSS = css({
-  display: 'flex'
+  display: 'flex',
+
+  '& > .sr-only': {
+    color: 'transparent',
+    height: '1px',
+    left: '-10000px',
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]:
```suggestion
    left: -10000,
```

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 3]
```diff
@@ -12,7 +12,17 @@ import MicrophoneIcon from './Assets/MicrophoneIcon';
 const { DictateState } = Constants;
 
 const ROOT_CSS = css({
-  display: 'flex'
+  display: 'flex',
+
+  '& > .sr-only': {
+    color: 'transparent',
+    height: '1px',
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]:
```suggestion
    height: 1,
```

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 4]
```diff
@@ -12,7 +12,17 @@ import MicrophoneIcon from './Assets/MicrophoneIcon';
...
+    left: '-10000px',
+    overflow: 'hidden',
+    position: 'absolute',
+    top: 'auto',
+    width: '1px'
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]:
```suggestion
    width: 1
```

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 5]
```diff
@@ -12,7 +12,17 @@ import MicrophoneIcon from './Assets/MicrophoneIcon';
 const { DictateState } = Constants;
 
 const ROOT_CSS = css({
-  display: 'flex'
+  display: 'flex',
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]: Can you check if we need `display: flex` here?

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 6]
```diff
@@ -48,19 +58,30 @@ export default connectMicrophoneButton(
...
         alt={ localize('Speak', language) }
         disabled={ disabled }
         onClick={ click }
       >
         <MicrophoneIcon />
       </IconButton>
+      <div 
+        aria-live="polite"
+        className="sr-only"
+        id="sr-Microphone"
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]: I think ID should be:

- camelCasing, no hyphens or underscores
- Namespaced, candidates
   - `webchatMicrophoneButton`
   - `webchatSendBoxMicrophoneButton`
   - (Think/dream about future collisions)

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 7]
```diff
@@ -48,19 +58,30 @@ export default connectMicrophoneButton(
...
+        tabIndex="-1"
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]: Do we need `tabIndex`? I think `<div>` is no tabbable.

If we need, prefer `tabIndex={ -1 }`.

File: packages/component/src/SendBox/MicrophoneButton.js [hunk 8]
```diff
@@ -48,19 +58,30 @@ export default connectMicrophoneButton(
...
+      >
+        { dictating ? localize('Microphone on', language) : localize('Microphone off', language) }
```
Comment (compulim) [inline packages/component/src/SendBox/MicrophoneButton.js]:
```suggestion
        { localize(dictating ? 'Microphone on' : 'Microphone off', language) }
```

File: packages/component/src/SendBox/MicrophoneButton.js (hunk 7 above)
Comment (corinagum) [reply packages/component/src/SendBox/MicrophoneButton.js]: You're right, no `tabIndex` necessary

File: packages/component/src/SendBox/MicrophoneButton.js (hunk 5 above)
Comment (corinagum) [reply packages/component/src/SendBox/MicrophoneButton.js]: Looks like it's needed in Edge, if it's not there a scrollbar appears on Web Chat (for the whole window)
Leaving as is!

**Pull request merged**